service-workers/service-worker/unregister-then-register-new-script.https.html

```diff
@@ -72,113 +72,54 @@
-                        'registration.active');
-          iframe.remove();
-          return with_iframe(scope);
-        })
-      .then(function(frame) {
-          assert_equals(
-              frame.contentWindow.navigator.serviceWorker.controller.scriptURL,
-              normalizeURL(worker_url),
-              'the original worker should control a new document');
-          frame.remove();
-          return registration.unregister();
-        })
-      .then(function() {
-          t.done();
-        })
-      .catch(unreached_rejection(t));
-}, 'Registering a new script URL that 404s does resurrect an ' +
-       'unregistered registration');
+  const iframe = await with_iframe(scope);
 
-async_test(function(t) {
-    var scope = 'resources/scope/unregister-then-register-reject-install-worker';
-    var iframe;
-    var registration;
+  await registration.unregister();
 
-    service_worker_unregister_and_register(t, worker_url, scope)
-      .then(function(r) {
-          registration = r;
-          return wait_for_state(t, r.installing, 'activated');
-        })
-      .then(function() {
-          return with_iframe(scope);
-        })
-      .then(function(frame) {
-          iframe = frame;
-          return registration.unregister();
-        })
-      .then(function() {
-          // Step 5.1 of Register clears the uninstall flag before firing
-          // the install event:
-          //
-          //  https://w3c.github.io/ServiceWorker/#register-algorithm
-          var promise = navigator.serviceWorker.register(
-              'resources/reject-install-worker.js', { scope: scope });
-          return promise;
-        })
-      .then(function(r) {
-          registration = r;
-          return wait_for_state(t, r.installing, 'redundant');
-        })
-      .then(function() {
-          assert_equals(registration.installing, null,
-                        'registration.installing');
-          assert_equals(registration.waiting, null,
-                        'registration.waiting');
-          assert_equals(registration.active.scriptURL, normalizeURL(worker_url),
-                        'registration.active');
-          iframe.remove();
-          return with_iframe(scope);
-        })
-      .then(function(frame) {
-          assert_equals(
-              frame.contentWindow.navigator.serviceWorker.controller.scriptURL,
-              normalizeURL(worker_url),
-              'the original worker should control a new document');
-          frame.remove();
-          return registration.unregister();
-        })
-      .then(function() {
-          t.done();
-        })
-      .catch(unreached_rejection(t));
-  }, 'Registering a new script URL that fails to install does resurrect ' +
-       'an unregistered registration');
+  try {
+    await navigator.serviceWorker.register('this-will-404', { scope });
+    assert_unreached('register should reject');
+  } catch (err) {}
```

Comment:
We also have promise_rejects() but I'm not sure which is actually easier to read. I guess promise_rejects lets you test the exception type.

Reply:
`promise_rejects` is better, as it also ensures it's a promise that rejects, and not a function that throws.